src/modules/hide_hypechat/index.js

```diff
@@ -0,0 +1,17 @@
+import {ChatFlags, PlatformTypes, SettingIds} from '../../constants.js';
+import settings from '../../settings.js';
+import {hasFlag} from '../../utils/flags.js';
+import {loadModuleForPlatforms} from '../../utils/modules.js';
+
+class HideHypeChatModule {
+  constructor() {
+    settings.on(`changed.${SettingIds.CHAT}`, () => this.load());
+    this.load();
```

Comment:
I would reverse this ordering, first load then attach the listener.

Reply:
Could we listen for a mutation for this button? and then based on the result hide it.